src/components/header/Nav/Nav.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react';
+import { getUser, getUserById } from '@/pages/api/api';
+import Profile from '../Profile/Profile';
+import styles from './Nav.module.css';
+import Link from 'next/link';
+import Image from 'next/image';
+
+interface Props {
+  className?: string | undefined;
+  setUserId?: (value: number) => void;
+  id?: number | undefined;

```

Comment:
undefined 타입이 포함되어 있는 경우 이렇게 유니언 타입으로 해줄 필요 없이, 아래처럼 타입이름 뒤에 `?` 옵셔널연산자를 붙여주시면 됩니다. 이외에도 이와 같은 부분들 있으면 모두 수정해주시면 좋을 것 같아요.

```suggestion
    id?: number;
```